src/main/kotlin/org/rust/cargo/toolchain/impl/CargoMetadata.kt

```diff
@@ -256,32 +256,42 @@ object CargoMetadata {
         )
     }
 
-    private fun Package.clean(fs: LocalFileSystem, isWorkspaceMember: Boolean): CargoWorkspaceData.Package? {
-        val root = checkNotNull(fs.refreshAndFindFileByPath(PathUtil.getParentPath(manifest_path))?.canonicalFile) {
-            "`cargo metadata` reported a package which does not exist at `$manifest_path`"
-        }
+    private fun Package.clean(isWorkspaceMember: Boolean): CargoWorkspaceData.Package? {
+        val rootCanonicalPath = toCanonicalPathSafe(PathUtil.getParentPath(manifest_path))
+        val rootUrl = LocalFileSystem.PROTOCOL_PREFIX + rootCanonicalPath
         return CargoWorkspaceData.Package(
             id,
-            root.url,
+            rootUrl,
             name,
             version,
-            targets.mapNotNull { it.clean(root) },
+            targets.mapNotNull { it.clean(rootCanonicalPath) },
             source,
             origin = if (isWorkspaceMember) PackageOrigin.WORKSPACE else PackageOrigin.TRANSITIVE_DEPENDENCY,
             edition = edition.cleanEdition()
         )
     }
 
-    private fun Target.clean(root: VirtualFile): CargoWorkspaceData.Target? {
-        val mainFile = root.findFileByMaybeRelativePath(src_path)?.canonicalFile
-        return mainFile?.let {
-            CargoWorkspaceData.Target(
-                it.url,
-                name,
-                makeTargetKind(cleanKind, cleanCrateTypes),
-                edition.cleanEdition(),
-                doctest = doctest ?: true
-            )
+    private fun Target.clean(rootPath: String): CargoWorkspaceData.Target? {
+        val crateRootPath = if (FileUtil.isAbsolute(src_path)) {
+            src_path
+        } else {
+            rootPath + File.separator + src_path
```

Comment:
`File.separator` differs on different OS. At the same time, VFS uses only Unix file separator.
So it seems that it breaks `org.rust.cargo.project.workspace.WorkspaceImpl#findTargetByCrateRoot` method